src/meatie/internal/cache.py

```diff
@@ -1,45 +1,56 @@
-#  Copyright 2023 The Meatie Authors. All rights reserved.
-#  Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
-
-import time as sys_time
+import time
+from collections import OrderedDict
 from dataclasses import dataclass
-from typing import Any
+from typing import Any, Optional
 
-from meatie import Duration, Time
 
-__all__ = ["Cache"]
+@dataclass
+class _Record:
+    value: Any
+    expires_at: float
 
 
 class Cache:
-    def __init__(self) -> None:
-        self.__storage: dict[str, _Record] = {}
+    def __init__(self, max_size: Optional[int] = None) -> None:
+        self._storage: OrderedDict[str, _Record] = OrderedDict()
+        self._max_size = max_size
+        self._time_provider = time.monotonic
 
     def load(self, key: str) -> Any:
-        current_time = sys_time.monotonic()
-        return self.load_at(current_time, key)
-
-    def load_at(self, time: Time, key: str) -> Any:
-        record_opt = self.__storage.get(key)
-        if record_opt is None:
+        """Load a value from the cache."""
+        record = self._storage.get(key)
+        if record is None:
             return None
 
-        if record_opt.expires_at < time:
-            del self.__storage[key]
+        if record.expires_at < self._time_provider():
+            del self._storage[key]
             return None
 
-        record_opt.last_accessed_at = time
-        return record_opt.value
-
-    def store(self, key: str, value: Any, ttl: Duration) -> None:
-        current_time = sys_time.monotonic()
-        self.store_at(current_time, key, value, ttl)
-
-    def store_at(self, time: Time, key: str, value: Any, ttl: Duration) -> Any:
-        self.__storage[key] = _Record(value=value, expires_at=time + ttl, last_accessed_at=time)
-
-
-@dataclass()
-class _Record:
-    value: Any
-    expires_at: Time
-    last_accessed_at: Time
+        # Move to end to mark as recently used
+        self._storage.move_to_end(key)
```

Comment:
I would avoid adding comments that repeat what the following statement is doing. I would use inline comments only to explain what is not evident from the line of code, for instance, some unclear assumptions, or surprising behavours.